doc/contributing.md

```diff
@@ -273,18 +203,232 @@ This installs your project in *editable* mode, meaning that changes made to the
+  a colorbar*. Bad: *how to add a label to the colorbar and use two different CPTs and
+  use subplots*.
+* Try to make the example as simple as possible. Good: use only commands that are
+  required to show the feature you want to highlight. Bad: use advanced/complex Python
+  features to make the code smaller.
+* Use a sample dataset from `pygmt.datasets` if you need to plot data. If a suitable
+  dataset isn't available, open an issue requesting one and we'll work together to add
+  it.
+* Add comments to explain things are aren't obvious from reading the code. Good: *Use a
+  Mercator projection and make the plot 15 centimeters wide*. Bad: *Draw coastlines and
+  plot the data*.
+* Describe the feature that you're showcasing and link to other relevant parts of the
+  documentation.
+* SI units should be used in the example code for gallery plots.
+
+### Contributing Tutorials
+
+The tutorials (the User Guide in the docs) are also built by sphinx-gallery from the
+`.py` files in the `examples/tutorials` folder of the repository. To add a new tutorial:
+
+* Include a `.py` file in the `examples/tutorials` folder on the base of the repository.
+* Write the tutorial in "notebook" style with code mixed with paragraphs explaining what
+  is being done. See the other tutorials for the format.
+* Include the tutorial in the table of contents of the documentation (side bar). Do this
+  by adding a line to the User Guide `toc` directive in `doc/index.rst`. Notice that the
+  file included is the `.rst` generated by sphinx-gallery.
+* Choose the most representative figure as the thumbnail figure by adding a comment line
+  `# sphinx_gallery_thumbnail_number = <fig_number>` to any place (usually at the top)
+  in the tutorial. The *fig_number* starts from 1.
+
+Guidelines for a good tutorial:
+
+* Each tutorial should focus on a particular set of tasks that a user might want to
+  accomplish: plotting grids, interpolation, configuring the frame, projections, etc.
+* The tutorial code should be as simple as possible. Avoid using advanced/complex Python
+  features or abbreviations.
+* Explain the options and features in as much detail as possible. The gallery has
+  concise examples while the tutorials are detailed and full of text.
+* SI units should be used in the example code for tutorial plots.
+
+Note that the `Figure.show()` function needs to be called for a plot to be inserted into
+the documentation.
+
+### Editing the API Documentation
+
+The API documentation is built from the docstrings in the Python `*.py` files under
+the `pygmt/src/` and `/pygmt/datasets/` folders. **All docstrings** should follow the
+[numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard).
+All functions/classes/methods should have docstrings with a full description of all
+arguments and return values.
+
+While the maximum line length for code is automatically set by *Black*, docstrings
```

Comment:
I will remove italics for black but think we should leave it as 'is automatically' because the style checks will prevent merging based on maximum line length.